tests/Unit/Evolution/VariableFixing/Test_ParameterizedDeleptonization.cpp

```diff
@@ -71,22 +84,63 @@ void test_variable_fixer(
 
   // check Ye values that will naturally decrease
   variable_fixer(&specific_internal_energy, &electron_fraction, &pressure,
-                 &specific_enthalpy, density, equation_of_state);
+                 &specific_enthalpy, &dummy_temperature, density,
+                 equation_of_state);
 
   const Scalar<DataVector> expected_electron_fraction{
       DataVector{0.49, 0.3077746733472282, 0.2}};
 
   CHECK_ITERABLE_APPROX(electron_fraction, expected_electron_fraction);
+  CHECK_ITERABLE_APPROX(dummy_temperature, expected_temperature);
 
   // check Ye values that will stay constant
   const Scalar<DataVector> expected_electron_fraction_low{
       electron_fraction_low};
   variable_fixer(&specific_internal_energy, &electron_fraction_low, &pressure,
-                 &specific_enthalpy, density, equation_of_state);
+                 &specific_enthalpy, &dummy_temperature, density,
+                 equation_of_state);
 
   CHECK_ITERABLE_APPROX(electron_fraction_low, expected_electron_fraction_low);
 }
 
+// 3D EoS version
+void test_variable_fixer(
+    const VariableFixing::ParameterizedDeleptonization& variable_fixer,
+    const EquationsOfState::EquationOfState<true, 3>& equation_of_state) {
+  const Scalar<DataVector> density{DataVector{3.0e-9, 3.7e-9, 4.0e-9}};
+  Scalar<DataVector> electron_fraction{DataVector{0.5, 0.5, 0.5}};
+  Scalar<DataVector> temperature{DataVector{1.0, 1.0, 1.0}};
+  Scalar<DataVector> initial_temperature = temperature;
+
+  // These should remain unchanged, sinced param delep can only make Ye drop
+  Scalar<DataVector> electron_fraction_low{DataVector{0.1, 0.15, 0.2}};
+  Scalar<DataVector> specific_internal_energy{DataVector{1.0, 2.0, 3.0}};
+  auto pressure = equation_of_state.pressure_from_density_and_energy(
+      density, specific_internal_energy, electron_fraction);
+  auto specific_enthalpy = hydro::relativistic_specific_enthalpy(
+      density, specific_internal_energy, pressure);
+
+  // check Ye values that will naturally decrease
+  variable_fixer(&specific_internal_energy, &electron_fraction, &pressure,
+                 &specific_enthalpy, &temperature, density, equation_of_state);
+
+  const Scalar<DataVector> expected_electron_fraction{
+      DataVector{0.49, 0.3077746733472282, 0.2}};
+
```

Comment:
Also check that the values you expect to change, change. E.g. epsilon is the same and T&p change or something